Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs

```diff
@@ -62,11 +62,20 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
                     return;
                 }
 
+                // Properly format baseUrlPrefix
+                if (!string.IsNullOrEmpty(baseUrlPrefix))
+                {
+                    if (!baseUrlPrefix.StartsWith("/", StringComparison.OrdinalIgnoreCase))
+                    {
+                        baseUrlPrefix = "/" + baseUrlPrefix;
+                    }
+                }
+
                 // Always redirect back to the default path if the base prefix is invalid or missing
                 _logger.LogDebug("Normalizing an URL at {LocalPath}", localPath);
 
-                var uri = new Uri(localPath);
-                var redirectUri = new Uri(baseUrlPrefix + "/" + _configuration[DefaultRedirectKey]);
+                var uri = new Uri(httpContext.Request.Scheme + "://" + httpContext.Request.Host + localPath, UriKind.Absolute);
```

Comment:
Isn't this what the UriBuilder class is for?

Reply:
I can do it with an uribuilder if you prefer that